zul/src/archive/web/js/zul/inp/Slider.js

```diff
@@ -349,7 +349,12 @@ zul.inp.Slider = zk.$extends(zul.Widget, {
 	},
 	_endDrag: function (dg) {
 		var widget = dg.control,
-			pos = widget._realpos();
+			pos = widget._realpos(),
+			step = widget.getStep();
+
+		if (step != -1) {
+			pos = pos - (pos - widget._minpos) % step;

```

Comment:
1. `pos -= (pos - widget._minpos) % step;`
2. You should consider reaching to max. 
    When the pos is reaching to max but it is not a multiple of step, it could never set to max.
    ex: min 0, max 10, step 3. You will turn 10 into 9.